test/storage/block_compactor_test.cpp

```diff
@@ -1,390 +1,240 @@
-        // No need to check the null case, because the arrow data structure shares the null bitmap with
-        // our table, and thus will always be equal
-        if (varlen == nullptr) continue;
-        auto size UNUSED_ATTRIBUTE = varlen->Size();
-        auto dict_code = arrow_metadata.GetColumnInfo(layout, id).Indices()[i];
-        // Safe to do plus 1, because length array will always have one more element
-        EXPECT_EQ(arrow_column.Offsets()[dict_code + 1] - arrow_column.Offsets()[dict_code], varlen->Size());
-        if (!varlen->IsInlined()) {
-          EXPECT_EQ(arrow_column.Values() + arrow_column.Offsets()[dict_code], varlen->Content());
-        } else {
-          EXPECT_EQ(
-              std::memcmp(varlen->Content(), arrow_column.Values() + arrow_column.Offsets()[dict_code], varlen->Size()),
-              0);
-        }
-      }
-    }
-
-    txn_manager.Commit(txn, transaction::TransactionUtil::EmptyCallback, nullptr);  // Commit: will be cleaned up by GC
-    delete[] buffer;
-
-    for (auto &entry : tuple_set) {
-      EXPECT_EQ(entry.second, 0);  // All tuples from the original block should have been accounted for.
-    }
-
-    for (auto &entry : tuples) delete[] reinterpret_cast<byte *>(entry.second);  // reclaim memory used for bookkeeping
-
-    gc.PerformGarbageCollection();
-    gc.PerformGarbageCollection();  // Second call to deallocate.
-    // Deallocate all the leftover gathered varlens
-    // No need to gather the ones still in the block because they are presumably all gathered
-    for (storage::col_id_t col_id : layout.AllColumns())
-      if (layout.IsVarlen(col_id)) arrow_metadata.GetColumnInfo(layout, col_id).Deallocate();
-    block_store_.Release(block);
+  txn_manager_->Commit(txn2, transaction::TransactionUtil::EmptyCallback, nullptr);
+
+  auto txn3 = txn_manager_->BeginTransaction();
+  std::unique_ptr<execution::exec::ExecutionContext> execCtx =
+      MakeExecCtx(db_oid, txn3, common::ManagedPointer(
+          catalog_->GetAccessor(common::ManagedPointer(txn3),
+              db_oid)
+      )); // Possible error -> Two pointers arent set (last 2 arguments not passed)
+  txn_manager_->Commit(txn3, transaction::TransactionUtil::EmptyCallback, nullptr);
+
+  col_id_t *col_oids = new col_id_t[1];
+  col_oids[0] = (col_id_t)1;
+  auto block = table_ptr->begin()->GetBlock();
+  execution::exec::ExecutionContext *exec = execCtx.get();
+
+  // Initialise block compactor and perform compaction
+  storage::BlockCompactor compactor(exec, col_oids, table_name.c_str());
+  compactor.PutInQueue(block);
+  transaction::DeferredActionManager *deferred_action_manager_ptr = deferred_action_manager.Get();
+  transaction::TransactionManager *txn_manager_ptr = txn_manager_.Get();
+  compactor.ProcessCompactionQueue(deferred_action_manager_ptr, txn_manager_ptr);
+
+  // Check for correctness of compaction
```

Comment:
I think it might be good to also check the arrow metadata after compaction so that the num_records, col_info are all correct.

Reply:
We are not using this test as of now, marking as resolved.